internal/pkg/awsiam/kubectl.go

```diff
@@ -0,0 +1,44 @@
+package awsiam
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const kubectl = "kubectl"
+
+// GetAllPods uses the kubectl installed on the host machine
+// Modifies the command context to add ./bin to path
+func GetAllPods(ctx context.Context, opts ...string) ([]corev1.Pod, error) {
```

Comment:
Don't need to change it, but couldn't we do this using the kubectl in `executables`?
we would just need to implement a new `Executable` that runs the magic to add `./bin` to the path
It seems more flexible and we don't duplicate code (plus the get pods method in `executables` already has unit tests

Reply:
So the main reason i rewrote this was because the `kubectl` in executables uses the cli tools docker image.
When i am running kubectl here I pass in the kubeconfig for aws-iam-auth which has an `exec` to use the aws-iam-authenticator client binary which is downloaded in to `./bin`
When running kubectl through `docker run` the container does not have the aws-iam-auth client in it.
I am open to changing this, but I couldnt get an easy solution for using the downloaded local binary with `docker run`?

Reply:
The `kubectl` in `executables` doesn't always use the cli-tools image, it depends on the dependency we inject
For example, in the cli, it's possible to disable the cli-tools image and use directly the binaries from the host, it's just another type implementing the `Executable` interface
I'm proposing doing something similar, implementing that interface in a way that allows you to add the `iam-authenticator` to the path